app-experimental.tex

```diff
@@ -1387,14 +1460,97 @@ \subsection{Points to Pair} % <<<
+    \item Check whether the isentry can load a CAP-WIDTH wide word.
+  \end{itemize}
+  \item \insnnoref{CJAURLP}: CHERI jump and unseal into register linked pair; this instruction has three operands of which one is explicit and the other two are implicit.
+  The explicit source register is the register index to the linked code capability.
+  The implicit source register is IDC holding the linked data capability.
+  IDC is also the implicit destination register.
+  This instruction unseals the data capability and stores it into IDC as well as unsealing the code capability and installing it into the PCC register.
+  The instruction does the following capability checks:
+  \begin{itemize}
+    \item Check whether the code capability is tagged.
+    \item Check whether the code capability is sealed with the \cotype{} $2^{64} - 7$.
+    \item Check whether the cursor of the code capability has at least two bytes in bounds.
+    \item Check whether the data capability is tagged.
+    \item Check whether the data capability is sealed with the \cotype{} $2^{64} - 7$.
+    \item Check whether the reservation is valid.
+  \end{itemize}
+\end{enumerate}
+
+``Points to pair'' isentries need a separate instruction to seal the capability.
+This instruction will use the \cotype{} $2^{64} - 6$.
+
+\subsubsection{Security model}
+
+The three linked instructions described above require careful state storing in the reservation in order to not break the underlying security model of CHERI.
+The code capability is sealed with the \cotype{} $2^{64} - 7$.
+Thus the capability contents can be read, but the capability cannot be dereferenced.
+The only way of dereferencing the code capability is via a \insnnoref{CJAURLP} instruction.
+A similar argument holds for the data capability.
+It cannot be derefenced other than through a \insnnoref{CJAURLP} instruction.
+Through the reservation, the code and data capability are linked and thus only the two capabilities from the same isentry can be invoked.
+
+A write to the destination register of the code capability or to IDC will invalidate the reservation.
+However, all reads are allowed.
+A second \insnnoref{CLILC} or \insnnoref{CLILD} is a write and thus will invalidate the reservation.
+Equally, if one of two -- either \insnnoref{CLILC} or \insnnoref{CLILD} -- reads from a different ``points to pair'' isentry, this will invalidate the reservation as well.
+
+%\subsubsection{Order of instructions}
+%
+%This section describes one way of implementing the three instructions and its implications.
+
+\subsection{Encoding of isentry instructions}
+
+We specified multiple instructions in this Chapter.
+Some of them exhibit similar functionality.
+Therefore, we propose to encode the following instructions in the same encoding space and use ``mode'' bits to distinguish between the respective behaviour.
+
+\begin{itemize}
+  \item \insnref{CSealEntry}, \insnnoref{CISealPCC}, and \insnnoref{CISealPair}
```

Comment:
Are you proposing to define `CISealPCC` and `CISealPair` as well? Do they intend to construct isentries similarly to `CSealEntry`? Would they modify the mode bits as described above?